my-app/src/pages/boardPage/boardPage.tsx

```diff
@@ -45,11 +55,50 @@ const BoardPage = () => {
         setIsCreate(true);
     }
 
+    function dragStartHandler(e: React.DragEvent<HTMLDivElement>, column: ColumnItemType): void {
+        setCurrentColumn(column);
+    }
+
+    function dragOverHandler(e: React.DragEvent<HTMLDivElement>): void {
+        e.preventDefault();
+       (e.target as HTMLDivElement).style.background = '#ece6e6'
+    }
+
+    function dragEndHandler(e: React.DragEvent<HTMLDivElement>): void {
+     (e.target as HTMLDivElement).style.background = ''
+    }
+
+    function dropHandler(e: React.DragEvent<HTMLDivElement>, column: ColumnItemType): void {
+        e.preventDefault();
+       dispatch(
+        sortColumnList([...columnList], column, currentColumn)
+       );
+      (e.target as HTMLDivElement).style.background = ''
+    }
+
+    const sortColumns = (a: ColumnItemType, b: ColumnItemType) => {
+      if(a.columnOrder > b.columnOrder) {
+        return 1;
+       } else {
+        return -1;
+       }
+      };
+
     return (
         <div className={styles.container}>
             <>
-              {columnList?.map((column, i) => (
-                <Column key={i} columnItem={column} />
+              {columnList?.sort(sortColumns).map((column, i) => (
```

Comment:
every time you render the component, you mutate the state with the `sort` method (it mutates the original array). As a result you get this error: 
<img width="550" alt="Screenshot 2023-02-15 at 21 24 05" src="https://user-images.githubusercontent.com/65244286/219146094-fca666d0-ced4-444d-84f3-78076f853a41.png">

all you need is to copy your columnList and not mutate it directly, so do it like this instead:  ` [...columnList]?.sort(*** your code ***)`

you can also check this stackoverflow link, it's about the similar issue: https://stackoverflow.com/questions/41051302/react-and-redux-uncaught-error-a-state-mutation-was-detected-between-dispatche